core/mixins.py

```diff
@@ -45,6 +46,29 @@ def get_context_data(self, *args, **kwargs):
         return super().get_context_data(page=self.page, *args, **kwargs)
 
 
+class GetCMSComponentMixin:
+    @cached_property
+    def cms_component(self):
+        response = cms_api_client.lookup_by_slug(
+            slug=self.component_slug,
+            language_code=translation.get_language(),
+            draft_token=self.request.GET.get('draft_token'),
+            service_name=cms.COMPONENTS,
+        )
+        return handle_cms_response_allow_404(response)
+
+    def get_context_data(self, *args, **kwargs):
+
+        component_is_rtl = 'ar' in str(
+            self.cms_component['meta']['languages']
+        ) and translation.get_language() == 'ar'
```

Comment:
we can let django do the hard work:
```
from django.utils import translation

activate_language = translation.get_language()
activate_language_is_bidi = translation.get_language_info(activate_language)['bidi']
component_supports_activate_language = activated_language in self.cms_component['meta']['languages']
component_is_rtl = (activate_language_is_bidi and component_supports_activate_language)
```
this will also work for any bidi language, not just ar :)

Reply:
i forgot this was a thing. i'll change it